### ISSUE TITLE
how to remove/Dispose instance from TinyIOC cointainer in nancy self hosting

### ISSUE THREAD

**Issue by frndxyz** [NancyFx/Nancy#1963]
i am using nancy self hosting. there i created some classes that resolved by passing different parameters. how can i find those classes using params and dispose it at runtime without terminate console app.

**Comment by phillip-haydon** [comment]
They should get disposed of automatically because they are transient. They would dispose when they lose their scope (the module)

**Comment by frndxyz** [comment]
i am not taking about instance that are using by webservers. but instances are created in console app that alive until app terminate. i want to forcefully dispose/remove them from container.

**Comment by phillip-haydon** [comment]
Its the same....

**Comment by frndxyz** [comment]
can you give me a example, i dont actually understand..

**Comment by horsdal** [comment]
@frndxyz I think it is the other way around, tbh. You need to provide some sample code and explain what isn't working for you. There is just not enough to go on here.

**Comment by frndxyz** [comment]
@horsdal ok this is an sample..

 _applicationContext.Container.Register<IPool, Pool>().AsMultiInstance(); register this

resolving instance using this function call with params. i have this type of multiple instance with diffrent params and also inside this class instance there are some other dependency resolving that same way.   this function is inside "objectfactory" class.
 public IPool GetPool(IPoolConfig poolConfig)
        {
            var @params = new NamedParameterOverloads
            {
                {"poolConfig", poolConfig},                
            };

            return _applicationContext.Container.Resolve<IPool>(@params);
        }

getting instance like this.

Ipool p= objectFactory.GetPool(config)

now i want to dispose this class and its dependencies uses inside it. "remember its not a web application that dispose instance when request fulfill" my application is consoleapp i want to dispose this instance at runtime.

Source = https://github.com/int6/CoiniumServ/blob/develop/src/CoiniumServ/Pools/Pool.cs

**Comment by phillip-haydon** [comment]
Either implement `IDisposable` and wrap your `GetPool` in a using statement so its disposed of when its done, or let Nancy dispose of it. 

If its a transient object, a new object is created on every call, and the objects will be disposed of when they lose their scope. When the Module is disposed of after a request has been completed, it will dispose of your object too. Unless you register it as a Singleton. 

If you're using it outside of a Nancy Module or in a long living class then you will need to manage its lifecycle yourself, most likely by implementing `IDisposable` and a `using` block.

**Comment by khellang** [comment]
I'm closing this. If you still feel you haven't gotten your answer, you might want to ask over at StackOverflow :smile:

**Comment by Corstiaan84** [comment]
I ran into a similar situation. I have a nancy web app that loads and disposes dependencies on a per-request basis. I rolled my own queue processor that calls the same code as the web app but runs a console app so the dependencies do not get disposed and re-instantiated. I ended up creating a callback action that is injected into the the worker from the console app and gets called when running a job finishes. In the callback I dispose the container.

Perhaps this inspires someone who comes across a similar situation. Also, if you guys have any feedback on this approach, I am all ears :-).
